### PR TITLE
add more maintainer file coverage

### PR DESCRIPTION
**Pull request by nashif**
Increase coverage of maintainer file and add two new keys for tags and tests associated with each area. This was originally part of #65257.

- tests: ieee802154: fix test identifier
- tests: usb: move bc12 tests under tests/drivers/usb
- MAINTAINERS: add 2 new keys: tags, tests
- MAINTAINERS: add related tests to areas
- MAINTAINERS: Cover as many files as possible
- lib: move utilities into own folder

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -1790,6 +2120,37 @@ Kernel:
     - tests/kernel/mem_protect/
   labels:
     - "area: Kernel"
+  tests:
+    - kernel
+
+Utilities:
+  status: maintained
```
Comment (aescolar) [inline MAINTAINERS.yml:2127]: don't we need a maintainer then?

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -2013,6 +2446,8 @@ Native POSIX/Sim and POSIX arch:
     - "area: native port"
   description: >-
     POSIX architecture and SOC, native_posix & native_sim boards, and related drivers
+  tests:
+    - boards.native_posix
```
Comment (aescolar) [inline MAINTAINERS.yml:2450]: could you name this board.native_sim instead? (given that we are deprecating the native_posix board..)

File: tests/lib/cmsis_dsp/complexmath/testcase.yaml [hunk 3]
```diff
@@ -7,7 +7,8 @@ tests:
       - sam_e70_xplained
       - mps2_an521
       - native_sim
-    tags: cmsis_dsp
+      - native_posix
```
Comment (aescolar) [inline tests/lib/cmsis_dsp/complexmath/testcase.yaml:10]: no need to add native_posix, native_sim covers it, and we will be removing it soon

**Review by aescolar — DISMISSED** [review]

File: tests/lib/cmsis_dsp/complexmath/testcase.yaml (hunk 3 above)
Comment (nashif) [reply tests/lib/cmsis_dsp/complexmath/testcase.yaml]: remove this commit as it became unrelated.

File: MAINTAINERS.yml (hunk 2 above)
Comment (nashif) [reply MAINTAINERS.yml]: yep, had this change for a while before the change to native_sim, change now.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**